azurelinuxagent/ga/update.py

```diff
@@ -980,6 +985,34 @@ def _ensure_firewall_rules_persisted(dst_ip):
             message=msg,
             log_event=False)
 
+    @staticmethod
+    def _add_nonroot_tcp_firewallrule_if_not_enabled(dst_ip):
+
+        update_handler = get_update_handler()
```

Comment:
This instance of UpdateHandler is not needed.

Also, with the current code it is not a good idea to create multiple instances of the UpdateHandler since there is some shared static state

Reply:
Addressed